src/main/java/service/impl/FruitServiceImpl.java

```diff
@@ -0,0 +1,25 @@
+package service.impl;
+
+import java.util.Map;
+import model.FruitTransaction;
+import service.FruitService;
+import strategy.OperationHandler;
+
+public class FruitServiceImpl implements FruitService {
+    private final Map<FruitTransaction.Operation, OperationHandler> operationHandlers;
+
+    public FruitServiceImpl(Map<FruitTransaction.Operation, OperationHandler> operationHandlers) {
+        this.operationHandlers = operationHandlers;
+    }
+
+    @Override
+    public void applyTransaction(FruitTransaction transaction) {
+        OperationHandler handler = operationHandlers.get(transaction.getOperation());
+        if (handler != null) {

```

Comment:
Using if-else constructs to handle operations is not recommended. Since you have a map of strategies, it's better to use the Strategy pattern effectively to eliminate the need for if-else or switch-case constructs.